src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -500,6 +502,7 @@ else if(entry.endsWith("*"))
      * Credentials can be passed e.g. to support running Jenkins behind a (reverse) proxy requiring authorization
      * FIXME: similar to hudson.remoting.Util.openURLConnection which is still used in hudson.remoting.Launcher
      */
+    @SuppressFBWarnings(value = "URLCONNECTION_SSRF_FD", justification = "Used by the agent for retrieving connection info from the server.")
```

Comment:
So what's the secbug it's complaining about here? Can it be guarded just in case?

Reply:
"Server-Side Request Forgery occur when a web server executes a request to a user supplied destination parameter that is not validated. Such vulnerabilities could allow an attacker to access internal services or to launch attacks from your web server. "

But, it's the agent requesting the JNLP file from the server from user / administrator supplied values. The rules in findsecbugs are clearly biased towards webservers. Some of them don't apply to launching the agent.